core/src/scene/stops.cpp

```diff
@@ -0,0 +1,84 @@
+#include "stops.h"
+#include "csscolorparser.hpp"
+#include "yaml-cpp/yaml.h"
+#include <cassert>
+#include <algorithm>
+
+#define RED(col) (col % 256)
+#define GRE(col) ((col >> 8) % 256)
+#define BLU(col) ((col >> 16) % 256)
+#define ALP(col) ((col >> 24) % 256)
+
+namespace Tangram {
+
+Stops::Stops(const YAML::Node& _node, bool _isColor) {
+
+    if (!_node.IsSequence()) { return; }
+
+    for (const auto& frameNode : _node) {
+        if (!frameNode.IsSequence() || frameNode.size() != 2) { continue; }
+        float key = frameNode[0].as<float>();
+        if (_isColor) {
+            // parse color from sequence or string
+            uint32_t color = 0;
+            YAML::Node colorNode = frameNode[1];
+            if (colorNode.IsScalar()) {
+                color = CSSColorParser::parse(colorNode.as<std::string>()).getInt();
+            } else if (colorNode.IsSequence() && colorNode.size() >= 3) {
+                float alpha = colorNode.size() > 3 ? colorNode[3].as<float>() : 1.f;
+                color = alpha * 255.;
+                color = (color << 8) + colorNode[2].as<float>() * 255.;
+                color = (color << 8) + colorNode[1].as<float>() * 255.;
+                color = (color << 8) + colorNode[0].as<float>() * 255.;
+            }
+            frames.emplace_back(key, color);
```

Comment:
Agreed, HSL would be great to support and the technique you linked looks even better! We just need RGBA interpolation to match current tangram-js behavior.